Source/Core/Duality/Components/Physics/RigidBody.cs

```diff
@@ -772,56 +777,60 @@ public bool PickShapes(Vector2 worldCoord, Vector2 size, List<ShapeInfo> pickedS
 		{
 			if (this.body == null) return false;
 
-			Vector2 fsWorldCoord = PhysicsUnit.LengthToPhysical * worldCoord;
-			FarseerPhysics.Collision.AABB fsWorldAABB = new FarseerPhysics.Collision.AABB(fsWorldCoord, PhysicsUnit.LengthToPhysical * (worldCoord + size));
-
-			int oldCount = pickedShapes.Count;
-			for (int i = 0; i < this.body.FixtureList.Count; i++)
+			PolygonShape boxShape = new PolygonShape(new Vertices(new List<Vector2>
 			{
-				Fixture f = this.body.FixtureList[i];
-				ShapeInfo s = f.UserData as ShapeInfo;
+				PhysicsUnit.LengthToPhysical * worldCoord,
+				PhysicsUnit.LengthToPhysical * new Vector2(worldCoord.X + size.X, worldCoord.Y),
+				PhysicsUnit.LengthToPhysical * (worldCoord + size),
+				PhysicsUnit.LengthToPhysical * new Vector2(worldCoord.X, worldCoord.Y + size.Y) 
+			}), 1);
+
+			FarseerPhysics.Common.Transform boxTransform = new FarseerPhysics.Common.Transform();
+			boxTransform.SetIdentity();
+			
+			return PickShapes(boxShape, boxTransform, pickedShapes);
```

Comment:
Please use `this.` for instance member access as well. Would merge anyway since it's really just a detail, so whoever of us gets to it first can do it.